share/functions/diff.fish

```diff
@@ -0,0 +1,6 @@
+# Use colours in diff output, if supported
+if command diff --color /dev/null /dev/null 2>&1
+    function diff
+        command diff --color $argv
```

Comment:
Great idea! Does it make sense to use `--color=auto` to be more explicit?

Reply:
I considered it, but it really feels like auto is just there so there's something to accompany --color=never and --color=always. I can make that change if you'd rather, though.

Reply:
The question is, is there a `diff(1)` implementation that accepts `--color` and interprets it as `--color=always` in the GNU variant?
The rationale being that `diff a b > some-file` should probably not add colors.
Which could happen if someone made `diff` a wrapper script to `git diff --no-index` though I don't really know what they ought to expect then.

For `grep` we use `--color=auto` and for `ls` we prefer it if available. So I'd feel safer with that, although it probably doesn't matter.

Reply:
I just checked through the BSDs, and none of them have a --color option (save for DragonFly, which uses the GNU diffutils). I'll add it, nonetheless, lest this derail into bikeshedding.

Reply:
(although that could be a good argument for adding it, since it futureproofs against other implementations adding differing behaviour)